cogs/tags.py

```diff
@@ -870,6 +871,33 @@ async def _tag_all_text_mode(self, ctx):
         fp = io.BytesIO(table.render().encode('utf-8'))
         await ctx.send(file=discord.File(fp, 'tags.txt'))
 
+    async def _tag_all_json_mode(self, ctx):
+        query = """SELECT ROW_TO_JSON(row, false)
+                   FROM(
+                           SELECT tag_lookup.id,
+                               tag_lookup.name,
+                               tag_lookup.owner_id,
+                               tags.uses,
+                               $2
+                               OR $3 = tag_lookup.owner_id AS "can_delete",
+                               LOWER(tag_lookup.name) <> LOWER(tags.name) AS "is_alias",
+                               tags.content
+                           FROM tag_lookup
+                               INNER JOIN tags ON tags.id = tag_lookup.tag_id
+                           WHERE tag_lookup.location_id = $1
+                           ORDER BY tags.uses DESC
+                    ) row;
+                """
+
+        bypass_owner_check = ctx.author.id == self.bot.owner_id or ctx.author.guild_permissions.manage_messages
+        rows = await ctx.db.fetch(query, ctx.guild.id, bypass_owner_check, ctx.author.id)
+        if not rows:
+            return await ctx.send('This server has no server-specific tags.')
+
+        for r in rows:
+            fp = io.BytesIO(r[0].encode())
+        await ctx.send(file=discord.File(fp, 'tags.txt'))
```

Comment:
Shouldn't the extension be .json?

Reply:
Absolutely. I thought I did it, but I guess I must've Ctrl + Z'd it